### PR TITLE
Feature/location

### PR DESCRIPTION
**Pull request by tkumor3**
https://github.com/user-attachments/assets/c1ba0f93-861b-4723-807a-45a46a6abbb6

### CODE REVIEW

File: src/Screens/Location.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+const Location = () => {
+  const [location, setLocation] = useState<LocationLib.LocationObject | null>(
+    null
+  );
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      let { status } = await LocationLib.requestForegroundPermissionsAsync();
```
Comment (mironiasty) [inline src/Screens/Location.tsx:14]: why `let`?

File: src/Screens/Location.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import * as LocationLib from "expo-location";
+
+const Location = () => {
+  const [location, setLocation] = useState<LocationLib.LocationObject | null>(
+    null
+  );
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+  useEffect(() => {
```
Comment (mironiasty) [inline src/Screens/Location.tsx:12]: Can you create hook that updates updates user location when it is changing?
```ts
const {location} = useCurrentLocation();
```

File: src/Screens/Location.tsx (hunk 1 above)
Comment (tkumor3) [reply src/Screens/Location.tsx:14]: linter should catch it 😱

File: src/hooks/useUserLocation.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  emitChange();
+});
+
+const requestPermissions = async () => {
+  const { status: foregroundStatus } =
+    await LocationLib.requestForegroundPermissionsAsync();
+  if (foregroundStatus !== "granted") {
+    throw new Error("Foreground permissions not granted");
+  }
+  await LocationLib.startLocationUpdatesAsync(LOCATION_TASK);
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply src/hooks/useUserLocation.ts:52]: It depends on what our aim is, it is possible to go trancing user location even in the background. Currently, I stop this task if there are no subscribers to my location store:
https://github.com/tkumor3/pokemon/pull/16/files#diff-6ae4c930535230da1d36c03846a08635c342a41bef3f5899118903d9ae7f44beR21

File: src/hooks/useUserLocation.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const location = useSyncExternalStore(
+    locationStore.subscribe,
+    locationStore.getSnapshot
+  );
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        await requestPermissions();
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply src/hooks/useUserLocation.ts:65]: it also execute: `await LocationLib.startLocationUpdatesAsync(LOCATION_TASK);` will change the name :)

File: src/hooks/useUserLocation.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  },
+};
+
+function emitChange() {
+  for (let listener of listeners) {
+    listener();
+  }
+}
+
+TaskManager.defineTask(LOCATION_TASK, ({ data: { locations }, error }: any) => {
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply src/hooks/useUserLocation.ts]: Fixed

File: src/hooks/useUserLocation.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+const requestPermissions = async () => {
+  const { status: foregroundStatus } =
+    await LocationLib.requestForegroundPermissionsAsync();
+  if (foregroundStatus !== "granted") {
+    throw new Error("Foreground permissions not granted");
+  }
+  await LocationLib.startLocationUpdatesAsync(LOCATION_TASK);
+};
+
+const useUserLocation = () => {
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply src/hooks/useUserLocation.ts:58]: The alternative solution is to use `setInverval` and call `getCurrentPositionAsync` and update the state when the location change

File: src/hooks/useUserLocation.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkumor3) [reply src/hooks/useUserLocation.ts:55]: another thing is, if it will work in the backend I will need to change the way of storing localizations

**Pull request closed without merging**